### ISSUE TITLE
Catastrophic backtracking of regexp match in getCloseTag

### ISSUE THREAD

**Issue by myfonj** [formulahendry/vscode-auto-close-tag#41]
Invoking auto close tag after lengthy tag (say, SVG `<shape d="(lots of data)" />`) either void/self-closing (`<.../>`) or with white-space before `>` hangs the extension:
```
<x 
0 1 2 3 4 5 6 7 8 9
0 1 2 3 4 5 6 7 8 9
0 1 2 3 4 5 6 7 8 9
0 1 2 3 4 5 6 7 8 9
0 1 2 3 4 5 6 7 8 9
0 1 2 3 4 5 6 7 8 9
0 1 2 3 4 5 6 7 8 9
0 1 2 3 4 5 6 7 8 9
0 1 2 3 4 5 6 7 8 9/>
<y>|
```
(try https://regex101.com/r/CZBpI4/1 )

I'd try to change the regexp at [1] from
` /<(\/?[a-zA-Z][a-zA-Z0-9:\-_.]*)(?:\s+[^<>]*?[^\s/<>=]+?)*?>/g`
(BTW, why such complicated?) to say
`<((\/?)[a-zA-Z][a-zA-Z0-9:\-_.]*)\s*[^>]*?(\/?)>`
to prevent that (just guess, haven't tested myself yet in VSC).

Sorry for hurried report; just burnt evening tracking this with no time left for fixup.

[1] https://github.com/formulahendry/vscode-auto-close-tag/blob/master/src/extension.ts#L135

---

What seems scary is that hung extension host survives restart of VS Code and continues eating CPU as a zombie process. Is it normal?

**Comment by formulahendry** [comment]
Hi @myfonj , thanks for reporting this with detailed repro steps! I will investigate it. Before making any big change on the regex, I will add some #43 test automation to make sure there will be no regression, As you may know, there are a lot lot of corner cases in the Auto Close Tag.

**Comment by myfonj** [comment]
Yup, this could never be 100% perfect. (You know, _"You will not 'parse' HTML with regexp"_ and so on.)

I'd suspect this or similar regexp issue could be the cause of other weird quirks, namely #34 (lots of white space in tag) and perhaps #31 (seems that when extension host hangs on one task, other extension are out of luck; at least that's what I observed).

---

TBH, in the meantime I'm switching to the competitive alternative; I suppose you have seen it.  Amusing to compare both approaches :] https://github.com/compulim/vscode-closetag/blob/master/findTags.js
